### PR TITLE
Rename get_raw to get_ref_unwrap and get_raw_checked to get_ref

### PR DESCRIPTION
**Pull request by gwenn**
And test `From<FromSqlError> for Error`

### CODE REVIEW

File: src/row.rs [hunk 1]
```diff
@@ -312,7 +312,7 @@ impl<'stmt> Row<'stmt> {
     ///
     /// Returns an `Error::InvalidColumnName` if `idx` is not a valid column
     /// name for this row.
-    pub fn get_raw_checked<I: RowIndex>(&self, idx: I) -> Result<ValueRef<'_>> {
+    pub fn try_get_raw<I: RowIndex>(&self, idx: I) -> Result<ValueRef<'_>> {
```
Comment (thomcc) [inline src/row.rs:315]: Hmm, for non-raw gets it's `Row::get` vs `Row::get_unwrap`. Maybe this should be `get_raw` and `get_raw_unwrap`?

Comment (gwenn) [reply src/row.rs:315]: It means that `get_raw_checked` becomes `get_raw` and `get_raw` becomes `get_raw_unwrap` !

Comment (gwenn) [reply src/row.rs:315]: Also:
https://docs.rs/postgres/0.18.1/postgres/row/struct.Row.html#method.try_get
https://docs.rs/sqlx/0.3.5/sqlx/row/trait.Row.html#method.try_get

Comment (thomcc) [reply src/row.rs:315]: Not sure I follow.

And IDK if sqlx/postgres are relevant — I think being consistent with non-raw gets is more important than following the names other crates use.

**Comment by dubiousjim** [comment]
To avoid the pain of changing the sig of `get_raw`, why don't you instead do something like:

* rename `get_raw_checked` to `get_ref` or some other new name
* rename `get_raw` to `get_ref_unwrap`
* deprecate `get_raw` and `get_raw_checked`

**Comment by thomcc** [comment]
Hmm, I don't exactly remember my hesitation here. Well, it's not like the current release isn't a major breaking change anyway, we should probably just land it as-is.

**Review by thomcc — APPROVED** [review]
LGTM as-is. Land unless there's something holding you back.

**Comment by gwenn** [comment]
> LGTM as-is. Land unless there's something holding you back.

For me, to ease migration of existing code, we should avoid doing both:
* renaming `old_method_name_1` to `new_method_name_1` (`get_raw` => `get_raw_unwrap`)
* renaming `old_method_name_2` to `old_method_name_1` (`get_raw_checked` => `get_raw`)
Because if you do first `s/\<get_raw_checked\>/\<get_raw/>/g` and then `s/\<get_raw\>/\<get_raw_unwrap/>/g` inadvertently, you are in trouble.

**Comment by thomcc** [comment]
Fair enough. The suggestion https://github.com/rusqlite/rusqlite/pull/838#issuecomment-747147992 works too.

**Pull request merged**